fw/http_frame.c

```diff
@@ -1967,23 +1973,30 @@ tfw_h2_insert_frame_header(struct sock *sk, TfwH2Ctx *ctx, TfwStream *stream,
 	length = tfw_h2_calc_frame_length(ctx, stream, type, len,
 					  max_len - FRAME_HEADER_SIZE);
 	if (type == HTTP2_DATA) {
-		ctx->rem_wnd -= length;
-		stream->rem_wnd -= length;
+		if (!stream->xmit.t_len) {
+			ctx->rem_wnd -= length;
+			stream->rem_wnd -= length;
+		}
 		stream->xmit.b_len -= length;
-	} else {
+	} else if (stream->xmit.h_len) {
 		stream->xmit.h_len -= length;
+	} else if (stream->xmit.t_len) {
+		ctx->rem_wnd -= length;
```

Comment:
Why we decrement window here? trailers are the headers frames.

Reply:
ditto.